drivers/scheduler/k8s/k8s.go

```diff
@@ -193,11 +193,17 @@ func (k *K8s) RefreshNodeRegistry() error {
 }
 
 //ParseSpecs parses the application spec file
-func (k *K8s) ParseSpecs(specDir string) ([]interface{}, error) {
+func (k *K8s) ParseSpecs(specDir, storageProvisioner string) ([]interface{}, error) {
 	fileList := make([]string, 0)
 	if err := filepath.Walk(specDir, func(path string, f os.FileInfo, err error) error {
 		if !f.IsDir() {
-			fileList = append(fileList, path)
+			if !isValidProvider(path) {
+				fileList = append(fileList, path)
+			} else { //specs from cloud provider directory
+				if strings.Contains(path, storageProvisioner) {
```

Comment:
Needs to be similar to the check in `isValidProvider`

Reply:
fixed.